src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs

```diff
@@ -250,10 +244,8 @@ internal static class MethodGenerator
                     {
                         if (localFunction)
                         {
-                            var preferStaticLocalFunction = workspace.Options.GetOption(CSharpCodeStyleOptions.PreferStaticLocalFunction).Value;
-
                             // Static local functions are only supported in C# 8.0 and later.
-                            if (preferStaticLocalFunction && languageVersion >= LanguageVersion.CSharp8)
+                            if (options.PreferStaticLocalFunction && languageVersion >= LanguageVersion.CSharp8)
```

Comment:
i think it's good that you're not passing in Workspace anymore.  Howeer, what i would do instead is just pass in the OptionSet so that we encapsulate all possible options we might need at these lower layers.

Reply:
I was thinking about that, but the problem I ran into was that all three code style options are C#-specific, so if we passed in the option set then I don't know if there would be a way to get the specific option we want.

Reply:
I'm not seeing the problem.  The code that needs to generate this stuff will be in C# land anyways, so it can check for the values of those options.